datalad/interface/save.py

```diff
@@ -68,60 +97,82 @@ class Save(Interface):
             args=("--version-tag", ),
             metavar='ID',
             doc="""an additional marker for that state.""",
-            constraints=EnsureStr() | EnsureNone()),)
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+    )
 
     @staticmethod
     @datasetmethod(name='save')
     def __call__(message=None, files=None, dataset=None,
-                 auto_add_changes=False, version_tag=None):
+                 auto_add_changes=False, version_tag=None,
+                 recursive=False, recursion_limit=None):
         # shortcut
         ds = require_dataset(dataset, check_installed=True,
                              purpose='saving')
 
-        _datalad_msg = False
-        if not message:
-            message = 'Recorded existing changes'
-            _datalad_msg = True
-
+        # use the dataset's base path to indiciate that everything
+        # should be saved
         if auto_add_changes:
             files = [ds.path]
 
-        # treat special case of still untracked subdatasets.
-        # those need to become submodules now, as they are otherwise added
-        # without an entry in .gitmodules, and subsequently break Git's
-        # submodule functionality completely
-        for utf in ds.repo.repo.untracked_files:
-            utf_abspath = opj(ds.path, utf)
-            if not isdir(utf_abspath):
-                # this cannot be a repository
-                continue
-
-            # test whether the potential submodule is scheduled for saving
-            utf_realpath = realpath(utf_abspath)
-            if any([utf_realpath.startswith(_with_sep(realpath(f)))
-                    for f in files]):
-                # matches at least one path -> turn into submodule
-                _install_subds_inplace(
-                    ds=ds,
-                    path=utf_abspath,  # can be ignored, we don't need the return value
-                    relativepath=utf,
-                    name=None)
+        # track whether we modified anything, so it becomes
+        # possible to decide when/what to save further down
+        # and one level up
+        _modified_flag = False
```

Comment:
I am never joking ;-)

(this line is the anchor for the comment, nothing more)